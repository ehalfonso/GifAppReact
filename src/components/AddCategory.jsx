import React, { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");
  const onInputChange = (e) => {
    setInputValue(e.target.value);
  };
  const onSubmit = (e) => {
    e.preventDefault();
    const cleanCategory = inputValue.trim();
    if (cleanCategory.length <= 1) return;
    setInputValue("");
    onNewCategory(cleanCategory);
  };
  return (
    <form aria-label="form" onSubmit={onSubmit}>
      <input
        type="text"
        value={inputValue}
        placeholder="shear gifs"
        onChange={onInputChange}
      />
    </form>
  );
};
AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
