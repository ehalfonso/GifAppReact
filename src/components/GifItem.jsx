import { PropTypes } from "prop-types";
export const GifItem = ({ title, url, id, remove }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
      <button onClick={() => remove(id)}>Eliminar</button>
    </div>
  );
};
GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};
