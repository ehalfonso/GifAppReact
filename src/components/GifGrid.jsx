import { GifItem } from "./GifItem";
import { useFecthGigfs } from "../hooks/useFecthGigfs";
import { useState } from "react";
import { PropTypes } from "prop-types";

export const GifGrid = ({ category }) => {
  const { images, isLoading, handleDelete } = useFecthGigfs(category);
  const removeSelect = (id) => {
    const newImges = images.filter((img) => id !== img.id);
    handleDelete(newImges);
  };
  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}
      <div className="card-grid">
        {images.map((img) => (
          <GifItem key={img.id} {...img} remove={removeSelect} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
