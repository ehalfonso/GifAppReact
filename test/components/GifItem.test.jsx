import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test in the <GifItem/>", () => {
  const title = "Saitama";
  const url = "https://one-pounch/saitama.jpg";
  const id = "Rd3jdna";
  test("should match with the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} id={id} />);
    expect(container).toMatchSnapshot();
  });
  test("should show the image with url and the alt ", () => {
    render(<GifItem title={title} url={url} id={id} />);
    //metodo para verificar que coincidad todo en el atributo img
    // expect(screen.getByRole("img").src).toBe(url);
    //el metodo anterior simplificado
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  test("should show the title in the component", () => {
    render(<GifItem title={title} url={url} id={id} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
