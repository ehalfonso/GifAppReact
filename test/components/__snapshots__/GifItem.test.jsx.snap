// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in the <GifItem/> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-pounch/saitama.jpg"
    />
    <p>
      Saitama
    </p>
    <button>
      Eliminar
    </button>
  </div>
</div>
`;
